src/test/java/unit/AwsS3StorageFolderControlled.java

```diff
@@ -0,0 +1,124 @@
+package unit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.cli.businessobject.Resource;
+import bio.terra.cli.serialization.userfacing.resource.UFAwsS3StorageFolder;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnitAws;
+import harness.utils.AwsS3StorageFolderUtils;
+import harness.utils.TestUtils;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Matchers;
+import org.json.JSONObject;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra resource` commands that handle controlled AWS S3 Storage Folders. */
+@Tag("unit-aws")
+public class AwsS3StorageFolderControlled extends SingleWorkspaceUnitAws {
+
+  @Test
+  @DisplayName(
+      "list, describe and resolve reflect creating and deleting a controlled storage folder")
+  void listDescribeResolveReflectCreateDelete() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getUserFacingId());
+
+    // `terra resource create s3-storage-folder --name=$storageFolderName`
+    String resourceName = UUID.randomUUID().toString();
+    UFAwsS3StorageFolder createdResource =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFAwsS3StorageFolder.class,
+            "resource",
+            "create",
+            "s3-storage-folder",
+            "--name=" + resourceName);
+
+    // check the created workspace has an id and aws details
+    assertNotNull(createdResource.bucketName, "create resource returned a aws bucket name");
+    assertNotNull(createdResource.prefix, "create resource returned a aws prefix");
+    assertEquals(resourceName, createdResource.name, "create resource resource name matches name");
+    assertEquals(createdResource.numObjects, 0, "create resource contains no objects");
```

Comment:
nit: swap arguments, assertEquals should be (expected, actual, [message])

Reply:
Fixed